release-please publish step fails: missing GH_TOKEN and malformed tag_name reference

The `Publish` step in `release-please.yml` fails when a release is created, so new versions are tagged/released on GitHub but never published to npm. Two bugs:

1. The `gh workflow run` command has no `GH_TOKEN` in its `env`, so the GitHub CLI exits with code 4 (`set the GH_TOKEN environment variable`).
2. `TAG_NAME` is set from `steps.release.outputs.outputs.tag_name` (doubled `outputs`), which resolves to empty. The correct expression is `steps.release.outputs.tag_name`.

## Motivation

Ensure a created release is automatically published to npm. Currently v1.6.3 was released on GitHub (tag `v1.6.3`) but npm `latest` is still `1.6.2`.

## Additional details

Failing run: https://github.com/mdn/mdn-http-observatory/actions/runs/29510236027/job/87661465017

```
env:
  TAG_NAME:                    # empty due to outputs.outputs typo
gh: To use GitHub CLI in a GitHub Actions workflow, set the GH_TOKEN environment variable.
##[error]Process completed with exit code 4.
```

Introduced in #506. Once fixed, v1.6.3 still needs a one-off manual `npm-publish.yml` dispatch with `tag: v1.6.3`.

--- a/.github/workflows/release-please.yml
+++ b/.github/workflows/release-please.yml
@@ -31,4 +31,5 @@ jobs:
             --ref "$TAG_NAME" \
             -f tag="$TAG_NAME"
         env:
-          TAG_NAME: ${{ steps.release.outputs.outputs.tag_name }}
+          GH_TOKEN: ${{ github.token }}
+          TAG_NAME: ${{ steps.release.outputs.tag_name }}

